test/synapse/2019-06-01/azartifacts/notebook.go

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package azartifacts
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+)
+
+// NotebookOperations contains the methods for the Notebook group.
+type NotebookOperations interface {
+	// BeginCreateOrUpdateNotebook - Creates or updates a Note Book.
+	BeginCreateOrUpdateNotebook(ctx context.Context, notebookName string, notebook NotebookResource, notebookCreateOrUpdateNotebookOptions *NotebookCreateOrUpdateNotebookOptions) (*NotebookResourcePollerResponse, error)
+	// ResumeCreateOrUpdateNotebook - Used to create a new instance of this poller from the resume token of a previous instance of this poller type.
+	ResumeCreateOrUpdateNotebook(token string) (NotebookResourcePoller, error)
+	// BeginDeleteNotebook - Deletes a Note book.
+	BeginDeleteNotebook(ctx context.Context, notebookName string) (*HTTPPollerResponse, error)
+	// ResumeDeleteNotebook - Used to create a new instance of this poller from the resume token of a previous instance of this poller type.
+	ResumeDeleteNotebook(token string) (HTTPPoller, error)
+	// GetNotebook - Gets a Note Book.
+	GetNotebook(ctx context.Context, notebookName string, notebookGetNotebookOptions *NotebookGetNotebookOptions) (*NotebookResourceResponse, error)
+	// GetNotebookSummaryByWorkSpace - Lists a summary of Notebooks.
+	GetNotebookSummaryByWorkSpace() (NotebookListResponsePager, error)
+	// GetNotebooksByWorkspace - Lists Notebooks.
+	GetNotebooksByWorkspace() (NotebookListResponsePager, error)
+}
+
+// notebookOperations implements the NotebookOperations interface.
+type notebookOperations struct {
+	*Client
+}
+
+// CreateOrUpdateNotebook - Creates or updates a Note Book.
+func (client *notebookOperations) BeginCreateOrUpdateNotebook(ctx context.Context, notebookName string, notebook NotebookResource, notebookCreateOrUpdateNotebookOptions *NotebookCreateOrUpdateNotebookOptions) (*NotebookResourcePollerResponse, error) {
+	return nil, nil
```

Comment:
@jhendrixMSFT FYI, this is how I'm outputting the operations at the moment. Or should I add an errors.New("Not yet implemented"). Processing and initializing a request for a data-plane poller is still TBD and the helper code is not being outputted.

Reply:
Probably best to emit it as `errors.New("NYI")` just to be safe.